benchmark/cmd/benchmark.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+
+	"k8s.io/perf-tests/benchmark/pkg/comparer"
+	"k8s.io/perf-tests/benchmark/pkg/runselector"
+	"k8s.io/perf-tests/benchmark/pkg/scraper"
+	"k8s.io/perf-tests/benchmark/pkg/util"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+)
+
+var (
+	leftJobName, rightJobName string
+	runSelectionScheme        string
+	nValue                    int
+	comparisonScheme          string
+	matchThreshold            float64
+)
+
+func registerFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&leftJobName, "left-job-name", "ci-kubernetes-e2e-gce-scalability", "Name of the job to be used as left hand side of comparison")
+	fs.StringVar(&rightJobName, "right-job-name", "ci-kubernetes-kubemark-100-gce", "Name of the job to be used as right hand side of comparison")
+	fs.StringVar(&runSelectionScheme, "run-selection-scheme", runselector.LastNHours, "Scheme for selecting the runs to be compared")
+	fs.IntVar(&nValue, "n", 24, "Value of 'n' (runs/hours depending on run-selection-scheme) to be used for run selection")
+	fs.StringVar(&comparisonScheme, "comparison-scheme", comparer.AvgTest, "Statistical test to be used as the algorithm for comparison")
+	fs.Float64Var(&matchThreshold, "match-threshold", 0.8, "The threshold for metric comparison, interpretation depends on test used (significance level for KSTest, bound for ratio of avgs in AvgTest)")
+}
+
+func main() {
+	// Set the tool's flags.
+	registerFlags(pflag.CommandLine)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	pflag.Parse()
+
+	// Select the runs of the left and right jobs to be used for comparison using the given run-selection scheme.
+	glog.Infof("Selecting runs for jobs '%v' and '%v' using scheme '%v' with n = %v\n", leftJobName, rightJobName, runSelectionScheme, nValue)
```

Comment:
I changed the name of the variable to `nForRunSelection`. However, I guess the log itself seems clear enough. It looks like:

```
Selecting runs for jobs 'ci-kubernetes-e2e-gce-scalability' and 'ci-kubernetes-kubemark-100-gce' using scheme 'last-n-hours' with n = 24
```

The context for 'n' comes from the scheme name 'last-n-hours'. Do you still suggest changing it?